daemon/prompter.c

```diff
@@ -574,46 +621,98 @@ prompter_finish_invocation(prompter_t *self)
 {
     GDBusMethodInvocation *invocation = self->prm_invocation;
     if( invocation ) {
+        change_cancellable(&self->prm_cancellable, NULL);
         self->prm_invocation = NULL;
 
-        uid_t uid = prompter_current_user(self);
+        if( !prompter_check_invocation(self, invocation) ) {
+            /* If we get here, the prompt was canceled */
+            prompter_return_error(invocation, G_DBUS_ERROR_AUTH_FAILED,
+                                  SERVICE_MESSAGE_NOT_ALLOWED);
+        }
 
-        appsettings_t *appsettings = NULL;
-        GVariant *parameters =
+        /* The prompt might have finished already
```

Comment:
"Might" is such an nasty word ;-) What is the uncertainty?

Reply:
This comment is in wrong place. It was supposed to be before `g_cancellable_cancel()` call, which was here before I made some changes and created that `change_cancellable()`.

The uncertainty came for the fact that it could be reached via properly finished windowprompt callback or via cancelled windowprompt and only in the latter case `prm_cancellable` would be `NULL`. Now I'm thinking that perhaps I should separate canceling and nullifying.

Reply:
Removed the comment. It was out of date.